src/containers/pages/PlanAssignment/index.tsx

```diff
@@ -0,0 +1,275 @@
+            // if hierarchyLimits is not populated we cannot proceed
+            handleBrokenPage(COULD_NOT_LOAD_PLAN_JURISDICTION_HIERARCHY);
+          }
+        }
+      })
+      .catch(e => {
+        handleBrokenPage(e.message);
+      });
+
+    // get all assignments
+    const OpenSrpAssignedService = new OpenSRPServiceClass(OPENSRP_GET_ASSIGNMENTS_ENDPOINT);
+    const assignmentsPromise = OpenSrpAssignedService.list({ plan: planId })
+      .then((response: any[]) => {
+        // TODO: remove any
+        if (response) {
+          const receivedAssignments: Assignment[] = response.map(assignment => {
+            return {
+              fromDate: moment(assignment.fromDate).format(),
+              jurisdiction: assignment.jurisdictionId,
+              organization: assignment.organizationId,
+              plan: assignment.planId,
+              toDate: moment(assignment.toDate).format(),
+            };
+          });
+          // save assignments to store
+          fetchAssignmentsActionCreator(receivedAssignments);
+        } else {
+          displayError(Error(COULD_NOT_LOAD_ASSIGNMENTS));
+        }
+      })
+      .catch(e => displayError(e));
+
+    // fetch all organizations
+    const OpenSRPOrganizationService = new OpenSRPServiceClass(OPENSRP_ORGANIZATION_ENDPOINT);
+    const organizationsPromise = OpenSRPOrganizationService.list()
+      .then((response: Organization[]) => {
+        if (response) {
+          // save all organizations to store
+          fetchOrganizationsActionCreator(response);
+        } else {
+          if (organizations.length < 1) {
+            // this means that we basically have not succeeded to make this call
+            // if organizations is not populated we cannot proceed
+            handleBrokenPage(COULD_NOT_LOAD_TEAMS);
+          }
+        }
+      })
+      .catch(e => {
+        handleBrokenPage(e.message);
+      });
+
+    // fetch current plan
+    const OpenSRPPlanService = new OpenSRPServiceClass(OPENSRP_PLANS);
+    const plansPromise = OpenSRPPlanService.read(planId)
+      .then((response: PlanDefinition[]) => {
+        if (response && response.length > 0) {
+          // save plan to store
+          addPlanActionCreator(response[0]);
+        } else {
+          if (!plan) {
+            // this means that we basically have not succeeded to make this call
+            // if no plan then we cannot proceed
+            handleBrokenPage(COULD_NOT_LOAD_PLAN);
+          }
+        }
+      })
+      .catch(e => {
+        handleBrokenPage(e.message);
+      });
+
+    Promise.all([planHierarchyPromise, assignmentsPromise, organizationsPromise, plansPromise])
+      .finally(() => setLoading(false))
+      .catch(e => {
+        handleBrokenPage(e.message);
+      });
+  }, []);
+
+  if (loading) {
+    // TODO: show message of what is actually loading
+    // return <Fragment>plan hierarchy loading...</Fragment>;
```

Comment:
just want to confirm that this is intentional, did not see as if we have an issue to address this, seems like it has been categorized as a nice to have in the Pr description.

Reply:
Removed this